rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -73,13 +73,13 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
     const std::string & topic_name,
     rmw_qos_profile_t qos_profile,
     rclcpp::IntraProcessBufferType buffer_type,
-    std::shared_ptr<SerializationBase> serializer)
-  : SubscriptionIntraProcessBase(topic_name, qos_profile),
+    std::shared_ptr<SerializationBase> serializer,
+    bool is_serialized = false)
+  : SubscriptionIntraProcessBase(topic_name, qos_profile, is_serialized),
     any_callback_(callback), serializer_(serializer)
   {
     if (!std::is_same<MessageT, CallbackMessageT>::value &&
-      !std::is_same<MessageT, rclcpp::SerializedMessage>::value &&
-      !std::is_same<CallbackMessageT, rcl_serialized_message_t>::value)
+      !std::is_base_of<rcl_serialized_message_t, MessageT>::value)
```

Comment:
I wasn't exactly sure why this check is happening. Also, I wouldn't compile if we check for the `CallbackMessageT`, which I believe is the right thing to check for though.